ONSdigital/eq-questionnaire-runner
# Placeholders should take into account answers on the path

### What is the context of this PR?
Currently, placeholders do not take into account whether answers are on the path. Historically, this wasn't needed as it is not the placeholders' job to work that out, however, in v3 we added a placeholder transform `first_non_empty_item` which requires us to check that the answers are on the path. 

### How to review 
See if the error is still occurs : 
  - Launch `mbs_0106`
  - Click Start Survey
  - Submit `No, I need to report for a different period`
  - Enter a valid date range i.e 1 May 2016 to 20 May 2016
  - You should see the question has piped in your date range.
  - Go back to the first question and change your answer to `Yes, I can report for this period`
  - The next question will still pipe in your custom date range, but it should have piped in the date range from the metadata. 

- Check unit tests pass
- Check type hinting comments 

### Checklist
